docs/howto/use-alternative-secrets-backend.rst

```diff
@@ -308,19 +308,14 @@ of the connection object.
 Roll your own secrets backend
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-A secrets backend is a subclass of :py:class:`airflow.secrets.BaseSecretsBackend`, and just has to implement the
-:py:meth:`~airflow.secrets.BaseSecretsBackend.get_connections` method.
+A secrets backend is a subclass of :py:class:`airflow.secrets.BaseSecretsBackend` and must implement either
```

Comment:
I had to read the current docs and the implementation of `BaseSecretsBackend` more than once to understand what's going on. I think it's written wrong in the current docs, so happy with this change.

The thing that confused me is the current docs say to implement `airflow.secrets.BaseSecretsBackend.get_conn_uri` or `get_connections`, but `get_conn_uri` should only return an (internal) URI given an Airflow connection id, not an actual Connection object, so I think it's up to the person implementing to decide to use this or not. Instead, the method that should be implemented is `get_connections`.

Can we simply write that? ("... and must implement :py:meth:`~airflow.secrets.BaseSecretsBackend.get_connections`.")

Reply:
if you implement `get_conn_uri` then you don`t need to implement `get_connections` and vice versa.   because `get_connections` in base class calls `get_conn_uri`

Reply:
Ah I see, so `get_conn_uri` for returning a connection as a string in URI format (e.g. `postgresql://user:pass@localhost:5432/mydb`), and `get_connections` for returning a list of connection objects.

Reply:
Might be worth mentioning this? What do you think? Other than that, looks good to me.

Reply:
i think it's ok as is now.

we have said "implement either get conn uri or get connections" -- i think that's all we really need to say here.

discussion of uri format and connection model details is done elsewhere.